File: ms-account/src/main/java/com/bantads/entity/read/AccountHistory.java
package com.bantads.entity.read;

import java.math.BigDecimal;
import java.util.Date;

public class AccountHistory {
    private Long historyId;
    private String accountNumber;
    private TransactionType type;
    private String originClientCPF;
    private String originClientName;
    private String destinationClientCPF;
    private String destinationClientName;
    private BigDecimal amount;
    private Date createdAt;
}

File: front/src/models/AccountHistory.ts
export type HistoryType = 'deposito' | 'saque' | 'transferencia';

export interface AccountHistory {
    id: number
    accountNumber: string
    type: HistoryType
    originClientCpf: string
    originClientName: string
    destinationClientCpf: string | null
    destinationClientName: string | null
    amount: number
    createdAt: Date
}